xrviz/dashboard.py

```diff
@@ -0,0 +1,93 @@
+import panel as pn
+import xarray as xr
+import hvplot.xarray
+from cartopy import crs as ccrs
+from .sigslot import SigSlot
+from .control import Control
+from .utils import convert_widget
+
+
+class Dashboard(SigSlot):
+    """
+    This section provides access to the complete generated dashboard,
+    consisting of all subsections.
+
+    Parameters
+    ----------
+    data: `xarray` instance: `DataSet` or `DataArray`
+           datset is used to initialize.
+
+    Attributes
+    ----------
+    panel: Displays the generated dashboard.
+    control: Provides access to the generated control panel.
+    index_selectors: Provides access to generated index_selectors.
+    plot: Plot button, upon click generates graph according to
+          kwargs selected in other sub-sections.
+    output: Provides access to generated graph.
+    """
+
+    def __init__(self, data):
+        super().__init__()
+        self.data = data
+        self.control = Control(self.data)
+        self.plot = pn.widgets.Button(name='Plot', width=200)
+        self.index_selectors = self.control.fields.index_selectors
+        self.output = pn.Row(pn.Spacer(name='Graph'))
+
+        self._register(self.plot, 'plot_clicked', 'clicks')
+        self.connect('plot_clicked', self.create_plot)
+        self.control.displayer.connect('variable_selected', self.check_is_plottable)
+
+        self.panel = pn.Column(self.control.panel,
+                               self.plot,
+                               self.output
+                               )
+        if isinstance(self.data, xr.Dataset):
+            self.plot.disabled = True
+        else:
+            if self.data.name in self.data.coords or len(self.data.dims) <= 1:
+                self.plot.disabled = True
+
+    def create_plot(self, *args):
+        kwargs = self.control.kwargs
+        var = kwargs['']
+        graph_opts = {}
+        graph_opts['x'] = kwargs['x']
```

Comment:
These dictionary assignments would be tidier as a single dictionary construction

Reply:
Corrected.